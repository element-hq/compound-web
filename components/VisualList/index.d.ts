export { VisualList } from './VisualList';
export { VisualListItem } from './VisualListItem';
//# sourceMappingURL=index.d.ts.map