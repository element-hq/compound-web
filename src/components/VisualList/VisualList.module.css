/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

.visual-list {
  display: flex;
  flex-direction: column;
  gap: var(--cpd-space-1x);
  margin: 0;
  padding: 0;
  list-style-type: none;
  border-radius: var(--cpd-space-3x);
  overflow: hidden;
}
