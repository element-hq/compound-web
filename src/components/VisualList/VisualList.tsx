/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import React, { type JSX, type PropsWithChildren } from "react";
import styles from "./VisualList.module.css";
import classNames from "classnames";

interface VisualListProps extends React.HTMLProps<HTMLUListElement> {
  /**
   * The CSS class name.
   */
  className?: string;
}

/**
 * A list component.
 */
export function VisualList({
  className,
  children,
  ...props
}: PropsWithChildren<VisualListProps>): JSX.Element {
  return (
    <ul className={classNames(styles["visual-list"], className)} {...props}>
      {children}
    </ul>
  );
}
