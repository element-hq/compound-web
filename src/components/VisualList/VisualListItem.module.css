/*
 * Copyright 2024 New Vector Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.visual-list-item {
  display: flex;
  gap: var(--cpd-space-3x);
  padding: var(--cpd-space-3x) var(--cpd-space-4x);
  background-color: var(--cpd-color-bg-subtle-secondary);
  font: var(--cpd-font-body-md-medium);
  align-items: center;
}

.visual-list-item-icon {
  flex-shrink: 0;
  color: var(--cpd-color-icon-secondary);
}

.visual-list-item-icon-success {
  color: var(--cpd-color-icon-success-primary);
}

.visual-list-item-icon-destructive {
  color: var(--cpd-color-icon-critical-primary);
}
