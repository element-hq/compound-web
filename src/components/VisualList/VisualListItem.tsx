/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import React, {
  type ComponentType,
  type JSX,
  type PropsWithChildren,
} from "react";
import styles from "./VisualListItem.module.css";
import classNames from "classnames";

interface VisualListItemProps extends React.HTMLProps<HTMLLIElement> {
  /**
   * The CSS class name.
   */
  className?: string;
  /**
   * The icon component.
   */
  Icon: ComponentType<React.SVGAttributes<SVGElement>>;

  success?: boolean;
  destructive?: boolean;
}

/**
 * A list component.
 */
export function VisualListItem({
  className,
  children,
  Icon,
  success = false,
  destructive = false,
  ...props
}: PropsWithChildren<VisualListItemProps>): JSX.Element {
  return (
    <li
      className={classNames(styles["visual-list-item"], className)}
      {...props}
    >
      <Icon
        className={classNames(styles["visual-list-item-icon"], {
          [styles["visual-list-item-icon-success"]]: success,
          [styles["visual-list-item-icon-destructive"]]: destructive,
        })}
        width="24px"
        height="24px"
        aria-hidden={true}
      />
      {children}
    </li>
  );
}
