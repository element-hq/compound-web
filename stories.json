{"v":3,"stories":{"actioncontrol--round":{"id":"actioncontrol--round","name":"Round","title":"ActionControl","importPath":"./src/components/ActionControl/ActionControl.stories.tsx","tags":["story"],"kind":"ActionControl","story":"Round","parameters":{"__id":"actioncontrol--round","docsOnly":false,"fileName":"./src/components/ActionControl/ActionControl.stories.tsx"}},"alert--docs":{"id":"alert--docs","title":"Alert","name":"Docs","importPath":"./src/components/Alert/Alert.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Alert","story":"Docs","parameters":{"__id":"alert--docs","docsOnly":true,"fileName":"./src/components/Alert/Alert.stories.tsx"}},"alert--success":{"id":"alert--success","name":"Success","title":"Alert","importPath":"./src/components/Alert/Alert.stories.tsx","tags":["autodocs","story"],"kind":"Alert","story":"Success","parameters":{"__id":"alert--success","docsOnly":false,"fileName":"./src/components/Alert/Alert.stories.tsx"}},"alert--critical":{"id":"alert--critical","name":"Critical","title":"Alert","importPath":"./src/components/Alert/Alert.stories.tsx","tags":["autodocs","story"],"kind":"Alert","story":"Critical","parameters":{"__id":"alert--critical","docsOnly":false,"fileName":"./src/components/Alert/Alert.stories.tsx"}},"alert--info":{"id":"alert--info","name":"Info","title":"Alert","importPath":"./src/components/Alert/Alert.stories.tsx","tags":["autodocs","story"],"kind":"Alert","story":"Info","parameters":{"__id":"alert--info","docsOnly":false,"fileName":"./src/components/Alert/Alert.stories.tsx"}},"alert--without-close":{"id":"alert--without-close","name":"Without Close","title":"Alert","importPath":"./src/components/Alert/Alert.stories.tsx","tags":["autodocs","story"],"kind":"Alert","story":"Without Close","parameters":{"__id":"alert--without-close","docsOnly":false,"fileName":"./src/components/Alert/Alert.stories.tsx"}},"avatar--docs":{"id":"avatar--docs","title":"Avatar","name":"Docs","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Avatar","story":"Docs","parameters":{"__id":"avatar--docs","docsOnly":true,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar--round":{"id":"avatar--round","name":"Round","title":"Avatar","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","story"],"kind":"Avatar","story":"Round","parameters":{"__id":"avatar--round","docsOnly":false,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar--square":{"id":"avatar--square","name":"Square","title":"Avatar","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","story"],"kind":"Avatar","story":"Square","parameters":{"__id":"avatar--square","docsOnly":false,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar--button":{"id":"avatar--button","name":"Button","title":"Avatar","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","story"],"kind":"Avatar","story":"Button","parameters":{"__id":"avatar--button","docsOnly":false,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar--no-image-fallback":{"id":"avatar--no-image-fallback","name":"No Image Fallback","title":"Avatar","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","story"],"kind":"Avatar","story":"No Image Fallback","parameters":{"__id":"avatar--no-image-fallback","docsOnly":false,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar--large-no-image-fallback":{"id":"avatar--large-no-image-fallback","name":"Large No Image Fallback","title":"Avatar","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","story"],"kind":"Avatar","story":"Large No Image Fallback","parameters":{"__id":"avatar--large-no-image-fallback","docsOnly":false,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar--all-avatars":{"id":"avatar--all-avatars","name":"All Avatars","title":"Avatar","importPath":"./src/components/Avatar/Avatar.stories.tsx","tags":["autodocs","story"],"kind":"Avatar","story":"All Avatars","parameters":{"__id":"avatar--all-avatars","docsOnly":false,"fileName":"./src/components/Avatar/Avatar.stories.tsx"}},"avatar-stackedavatar--docs":{"id":"avatar-stackedavatar--docs","title":"Avatar/StackedAvatar","name":"Docs","importPath":"./src/components/Avatar/AvatarStack.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Avatar/StackedAvatar","story":"Docs","parameters":{"__id":"avatar-stackedavatar--docs","docsOnly":true,"fileName":"./src/components/Avatar/AvatarStack.stories.tsx"}},"avatar-stackedavatar--avatar-stack":{"id":"avatar-stackedavatar--avatar-stack","name":"Avatar Stack","title":"Avatar/StackedAvatar","importPath":"./src/components/Avatar/AvatarStack.stories.tsx","tags":["autodocs","story"],"kind":"Avatar/StackedAvatar","story":"Avatar Stack","parameters":{"__id":"avatar-stackedavatar--avatar-stack","docsOnly":false,"fileName":"./src/components/Avatar/AvatarStack.stories.tsx"}},"badge--badge":{"id":"badge--badge","name":"Badge","title":"Badge","importPath":"./src/components/Badge/Badge.stories.tsx","tags":["story"],"kind":"Badge","story":"Badge","parameters":{"__id":"badge--badge","docsOnly":false,"fileName":"./src/components/Badge/Badge.stories.tsx"}},"button--docs":{"id":"button--docs","title":"Button","name":"Docs","importPath":"./src/components/Button/Button.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Button","story":"Docs","parameters":{"__id":"button--docs","docsOnly":true,"fileName":"./src/components/Button/Button.stories.tsx"}},"button--primary":{"id":"button--primary","name":"Primary","title":"Button","importPath":"./src/components/Button/Button.stories.tsx","tags":["autodocs","story"],"kind":"Button","story":"Primary","parameters":{"__id":"button--primary","docsOnly":false,"fileName":"./src/components/Button/Button.stories.tsx"}},"button--secondary":{"id":"button--secondary","name":"Secondary","title":"Button","importPath":"./src/components/Button/Button.stories.tsx","tags":["autodocs","story"],"kind":"Button","story":"Secondary","parameters":{"__id":"button--secondary","docsOnly":false,"fileName":"./src/components/Button/Button.stories.tsx"}},"button--tertiary":{"id":"button--tertiary","name":"Tertiary","title":"Button","importPath":"./src/components/Button/Button.stories.tsx","tags":["autodocs","story"],"kind":"Button","story":"Tertiary","parameters":{"__id":"button--tertiary","docsOnly":false,"fileName":"./src/components/Button/Button.stories.tsx"}},"button--destructive":{"id":"button--destructive","name":"Destructive","title":"Button","importPath":"./src/components/Button/Button.stories.tsx","tags":["autodocs","story"],"kind":"Button","story":"Destructive","parameters":{"__id":"button--destructive","docsOnly":false,"fileName":"./src/components/Button/Button.stories.tsx"}},"checkbox--docs":{"id":"checkbox--docs","title":"Checkbox","name":"Docs","importPath":"./src/components/Checkbox/Checkbox.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Checkbox","story":"Docs","parameters":{"__id":"checkbox--docs","docsOnly":true,"fileName":"./src/components/Checkbox/Checkbox.stories.tsx"}},"checkbox--primary":{"id":"checkbox--primary","name":"Primary","title":"Checkbox","importPath":"./src/components/Checkbox/Checkbox.stories.tsx","tags":["autodocs","story"],"kind":"Checkbox","story":"Primary","parameters":{"__id":"checkbox--primary","docsOnly":false,"fileName":"./src/components/Checkbox/Checkbox.stories.tsx"}},"checkbox--critical":{"id":"checkbox--critical","name":"Critical","title":"Checkbox","importPath":"./src/components/Checkbox/Checkbox.stories.tsx","tags":["autodocs","story"],"kind":"Checkbox","story":"Critical","parameters":{"__id":"checkbox--critical","docsOnly":false,"fileName":"./src/components/Checkbox/Checkbox.stories.tsx"}},"checkbox--disabled":{"id":"checkbox--disabled","name":"Disabled","title":"Checkbox","importPath":"./src/components/Checkbox/Checkbox.stories.tsx","tags":["autodocs","story"],"kind":"Checkbox","story":"Disabled","parameters":{"__id":"checkbox--disabled","docsOnly":false,"fileName":"./src/components/Checkbox/Checkbox.stories.tsx"}},"checkbox--disabled-checked":{"id":"checkbox--disabled-checked","name":"Disabled Checked","title":"Checkbox","importPath":"./src/components/Checkbox/Checkbox.stories.tsx","tags":["autodocs","story"],"kind":"Checkbox","story":"Disabled Checked","parameters":{"__id":"checkbox--disabled-checked","docsOnly":false,"fileName":"./src/components/Checkbox/Checkbox.stories.tsx"}},"form--root":{"id":"form--root","name":"Root","title":"Form","importPath":"./src/components/Form/Form.stories.tsx","tags":["story"],"kind":"Form","story":"Root","parameters":{"__id":"form--root","docsOnly":false,"fileName":"./src/components/Form/Form.stories.tsx"}},"glass--docs":{"id":"glass--docs","title":"Glass","name":"Docs","importPath":"./src/components/Glass/Glass.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Glass","story":"Docs","parameters":{"__id":"glass--docs","docsOnly":true,"fileName":"./src/components/Glass/Glass.stories.tsx"}},"glass--glass":{"id":"glass--glass","name":"Glass","title":"Glass","importPath":"./src/components/Glass/Glass.stories.tsx","tags":["autodocs","story"],"kind":"Glass","story":"Glass","parameters":{"__id":"glass--glass","docsOnly":false,"fileName":"./src/components/Glass/Glass.stories.tsx"}},"icon--icon":{"id":"icon--icon","name":"Icon","title":"Icon","importPath":"./src/components/Icon/Icon.stories.tsx","tags":["story"],"kind":"Icon","story":"Icon","parameters":{"__id":"icon--icon","docsOnly":false,"fileName":"./src/components/Icon/Icon.stories.tsx"}},"button-iconbutton--docs":{"id":"button-iconbutton--docs","title":"Button/IconButton","name":"Docs","importPath":"./src/components/IconButton/IconButton.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Button/IconButton","story":"Docs","parameters":{"__id":"button-iconbutton--docs","docsOnly":true,"fileName":"./src/components/IconButton/IconButton.stories.tsx"}},"button-iconbutton--normal":{"id":"button-iconbutton--normal","name":"Normal","title":"Button/IconButton","importPath":"./src/components/IconButton/IconButton.stories.tsx","tags":["autodocs","story"],"kind":"Button/IconButton","story":"Normal","parameters":{"__id":"button-iconbutton--normal","docsOnly":false,"fileName":"./src/components/IconButton/IconButton.stories.tsx"}},"button-iconbutton--with-indicator":{"id":"button-iconbutton--with-indicator","name":"With Indicator","title":"Button/IconButton","importPath":"./src/components/IconButton/IconButton.stories.tsx","tags":["autodocs","story"],"kind":"Button/IconButton","story":"With Indicator","parameters":{"__id":"button-iconbutton--with-indicator","docsOnly":false,"fileName":"./src/components/IconButton/IconButton.stories.tsx"}},"button-iconbutton--with-highlight-indicator":{"id":"button-iconbutton--with-highlight-indicator","name":"With Highlight Indicator","title":"Button/IconButton","importPath":"./src/components/IconButton/IconButton.stories.tsx","tags":["autodocs","story"],"kind":"Button/IconButton","story":"With Highlight Indicator","parameters":{"__id":"button-iconbutton--with-highlight-indicator","docsOnly":false,"fileName":"./src/components/IconButton/IconButton.stories.tsx"}},"link--docs":{"id":"link--docs","title":"Link","name":"Docs","importPath":"./src/components/Link/Link.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Link","story":"Docs","parameters":{"__id":"link--docs","docsOnly":true,"fileName":"./src/components/Link/Link.stories.tsx"}},"link--round":{"id":"link--round","name":"Round","title":"Link","importPath":"./src/components/Link/Link.stories.tsx","tags":["autodocs","story"],"kind":"Link","story":"Round","parameters":{"__id":"link--round","docsOnly":false,"fileName":"./src/components/Link/Link.stories.tsx"}},"menuitem--docs":{"id":"menuitem--docs","title":"MenuItem","name":"Docs","importPath":"./src/components/MenuItem/MenuItem.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"MenuItem","story":"Docs","parameters":{"__id":"menuitem--docs","docsOnly":true,"fileName":"./src/components/MenuItem/MenuItem.stories.tsx"}},"menuitem--primary":{"id":"menuitem--primary","name":"Primary","title":"MenuItem","importPath":"./src/components/MenuItem/MenuItem.stories.tsx","tags":["autodocs","story"],"kind":"MenuItem","story":"Primary","parameters":{"__id":"menuitem--primary","docsOnly":false,"fileName":"./src/components/MenuItem/MenuItem.stories.tsx"}},"menuitem--critical":{"id":"menuitem--critical","name":"Critical","title":"MenuItem","importPath":"./src/components/MenuItem/MenuItem.stories.tsx","tags":["autodocs","story"],"kind":"MenuItem","story":"Critical","parameters":{"__id":"menuitem--critical","docsOnly":false,"fileName":"./src/components/MenuItem/MenuItem.stories.tsx"}},"togglemenuitem--docs":{"id":"togglemenuitem--docs","title":"ToggleMenuItem","name":"Docs","importPath":"./src/components/MenuItem/ToggleMenuItem.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"ToggleMenuItem","story":"Docs","parameters":{"__id":"togglemenuitem--docs","docsOnly":true,"fileName":"./src/components/MenuItem/ToggleMenuItem.stories.tsx"}},"togglemenuitem--primary":{"id":"togglemenuitem--primary","name":"Primary","title":"ToggleMenuItem","importPath":"./src/components/MenuItem/ToggleMenuItem.stories.tsx","tags":["autodocs","story"],"kind":"ToggleMenuItem","story":"Primary","parameters":{"__id":"togglemenuitem--primary","docsOnly":false,"fileName":"./src/components/MenuItem/ToggleMenuItem.stories.tsx"}},"radio--docs":{"id":"radio--docs","title":"Radio","name":"Docs","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Radio","story":"Docs","parameters":{"__id":"radio--docs","docsOnly":true,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"radio--primary":{"id":"radio--primary","name":"Primary","title":"Radio","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","story"],"kind":"Radio","story":"Primary","parameters":{"__id":"radio--primary","docsOnly":false,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"radio--checked":{"id":"radio--checked","name":"Checked","title":"Radio","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","story"],"kind":"Radio","story":"Checked","parameters":{"__id":"radio--checked","docsOnly":false,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"radio--checked-disabled":{"id":"radio--checked-disabled","name":"Checked Disabled","title":"Radio","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","story"],"kind":"Radio","story":"Checked Disabled","parameters":{"__id":"radio--checked-disabled","docsOnly":false,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"radio--critical":{"id":"radio--critical","name":"Critical","title":"Radio","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","story"],"kind":"Radio","story":"Critical","parameters":{"__id":"radio--critical","docsOnly":false,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"radio--critical-checked":{"id":"radio--critical-checked","name":"Critical Checked","title":"Radio","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","story"],"kind":"Radio","story":"Critical Checked","parameters":{"__id":"radio--critical-checked","docsOnly":false,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"radio--critical-checked-disabled":{"id":"radio--critical-checked-disabled","name":"Critical Checked Disabled","title":"Radio","importPath":"./src/components/Radio/Radio.stories.tsx","tags":["autodocs","story"],"kind":"Radio","story":"Critical Checked Disabled","parameters":{"__id":"radio--critical-checked-disabled","docsOnly":false,"fileName":"./src/components/Radio/Radio.stories.tsx"}},"search--docs":{"id":"search--docs","title":"Search","name":"Docs","importPath":"./src/components/Search/Search.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Search","story":"Docs","parameters":{"__id":"search--docs","docsOnly":true,"fileName":"./src/components/Search/Search.stories.tsx"}},"search--search":{"id":"search--search","name":"Search","title":"Search","importPath":"./src/components/Search/Search.stories.tsx","tags":["autodocs","story"],"kind":"Search","story":"Search","parameters":{"__id":"search--search","docsOnly":false,"fileName":"./src/components/Search/Search.stories.tsx"}},"toggle--docs":{"id":"toggle--docs","title":"Toggle","name":"Docs","importPath":"./src/components/Toggle/Toggle.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Toggle","story":"Docs","parameters":{"__id":"toggle--docs","docsOnly":true,"fileName":"./src/components/Toggle/Toggle.stories.tsx"}},"toggle--active":{"id":"toggle--active","name":"Active","title":"Toggle","importPath":"./src/components/Toggle/Toggle.stories.tsx","tags":["autodocs","story"],"kind":"Toggle","story":"Active","parameters":{"__id":"toggle--active","docsOnly":false,"fileName":"./src/components/Toggle/Toggle.stories.tsx"}},"toggle--checked":{"id":"toggle--checked","name":"Checked","title":"Toggle","importPath":"./src/components/Toggle/Toggle.stories.tsx","tags":["autodocs","story"],"kind":"Toggle","story":"Checked","parameters":{"__id":"toggle--checked","docsOnly":false,"fileName":"./src/components/Toggle/Toggle.stories.tsx"}},"toggle--disabled":{"id":"toggle--disabled","name":"Disabled","title":"Toggle","importPath":"./src/components/Toggle/Toggle.stories.tsx","tags":["autodocs","story"],"kind":"Toggle","story":"Disabled","parameters":{"__id":"toggle--disabled","docsOnly":false,"fileName":"./src/components/Toggle/Toggle.stories.tsx"}},"toggle--disabled-checked":{"id":"toggle--disabled-checked","name":"Disabled Checked","title":"Toggle","importPath":"./src/components/Toggle/Toggle.stories.tsx","tags":["autodocs","story"],"kind":"Toggle","story":"Disabled Checked","parameters":{"__id":"toggle--disabled-checked","docsOnly":false,"fileName":"./src/components/Toggle/Toggle.stories.tsx"}},"tooltip--docs":{"id":"tooltip--docs","title":"Tooltip","name":"Docs","importPath":"./src/components/Tooltip/Tooltip.stories.tsx","tags":["autodocs","docs"],"storiesImports":[],"kind":"Tooltip","story":"Docs","parameters":{"__id":"tooltip--docs","docsOnly":true,"fileName":"./src/components/Tooltip/Tooltip.stories.tsx"}},"tooltip--side":{"id":"tooltip--side","name":"Side","title":"Tooltip","importPath":"./src/components/Tooltip/Tooltip.stories.tsx","tags":["autodocs","story"],"kind":"Tooltip","story":"Side","parameters":{"__id":"tooltip--side","docsOnly":false,"fileName":"./src/components/Tooltip/Tooltip.stories.tsx"}},"tooltip--align":{"id":"tooltip--align","name":"Align","title":"Tooltip","importPath":"./src/components/Tooltip/Tooltip.stories.tsx","tags":["autodocs","story"],"kind":"Tooltip","story":"Align","parameters":{"__id":"tooltip--align","docsOnly":false,"fileName":"./src/components/Tooltip/Tooltip.stories.tsx"}},"typography--heading":{"id":"typography--heading","name":"Heading","title":"Typography","importPath":"./src/components/Typography/Heading.stories.tsx","tags":["story"],"kind":"Typography","story":"Heading","parameters":{"__id":"typography--heading","docsOnly":false,"fileName":"./src/components/Typography/Heading.stories.tsx"}},"typography--text":{"id":"typography--text","name":"Text","title":"Typography","importPath":"./src/components/Typography/Text.stories.tsx","tags":["story"],"kind":"Typography","story":"Text","parameters":{"__id":"typography--text","docsOnly":false,"fileName":"./src/components/Typography/Text.stories.tsx"}}}}
