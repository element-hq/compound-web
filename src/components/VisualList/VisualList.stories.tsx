/*
 * Copyright 2024 New Vector Ltd.
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */

import { VisualList as VisualListComponent } from "./VisualList";
import { type Meta, type StoryFn } from "@storybook/react-vite";
import React, { type ComponentProps } from "react";
import InfoIcon from "@vector-im/compound-design-tokens/assets/web/icons/info";
import { VisualListItem } from "./VisualListItem.tsx";

export default {
  title: "VisualList/VisualList",
  component: VisualListComponent,
  tags: ["autodocs"],
  argTypes: {},
  args: {
    children: (
      <>
        <VisualListItem Icon={InfoIcon}>First item</VisualListItem>
        <VisualListItem Icon={InfoIcon}>Second item</VisualListItem>
        <VisualListItem Icon={InfoIcon}>Third item</VisualListItem>
      </>
    ),
  },
} as Meta<typeof VisualListComponent>;

const Template: StoryFn<typeof VisualListComponent> = (
  args: ComponentProps<typeof VisualListComponent>,
) => {
  return <VisualListComponent {...args} />;
};

export const Default = Template.bind({});
Default.args = {};
