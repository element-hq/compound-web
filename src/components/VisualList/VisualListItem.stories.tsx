/*
 * Copyright 2024 New Vector Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { VisualListItem as VisualListItemComponent } from "./VisualListItem";
import { Meta, StoryFn } from "@storybook/react";
import React, { ComponentProps } from "react";
import InfoIcon from "@vector-im/compound-design-tokens/assets/web/icons/info";

export default {
  title: "VisualList/VisualListItem",
  component: VisualListItemComponent,
  tags: ["autodocs"],
  argTypes: {},
  args: {
    Icon: InfoIcon,
    children: "List item description",
  },
} as Meta<typeof VisualListItemComponent>;

const Template: StoryFn<typeof VisualListItemComponent> = (
  args: ComponentProps<typeof VisualListItemComponent>,
) => {
  return <VisualListItemComponent {...args} />;
};

export const Default = Template.bind({});
Default.args = {};

export const Success = Template.bind({});
Success.args = {
  success: true,
};

export const Destructive = Template.bind({});
Destructive.args = {
  destructive: true,
};

export const Multiline = Template.bind({});
Multiline.args = {
  children:
    "List item with a looooooooooong very looooooooooong loooooooooooooong description",
};
