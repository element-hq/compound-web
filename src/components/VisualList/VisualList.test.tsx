/*
 * Copyright 2024 New Vector Ltd.
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */

import { composeStories } from "@storybook/react";
import * as stories from "./VisualList.stories";
import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import React from "react";

const { Default } = composeStories(stories);

describe("VisualList", () => {
  it("renders a Default VisualList", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
