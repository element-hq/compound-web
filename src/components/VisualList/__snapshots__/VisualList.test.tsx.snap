// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VisualList > renders a Default VisualList 1`] = `
<div>
  <ul
    class="_visual-list_7f22f8"
  >
    <li
      class="_visual-list-item_1553df"
    >
      <svg
        aria-hidden="true"
        class="_visual-list-item-icon_1553df"
        fill="currentColor"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.288 7.288A.97.97 0 0 1 12 7q.424 0 .713.287Q13 7.576 13 8t-.287.713A.97.97 0 0 1 12 9a.97.97 0 0 1-.713-.287A.97.97 0 0 1 11 8q0-.424.287-.713m.001 4.001A.97.97 0 0 1 12 11q.424 0 .713.287.287.288.287.713v4q0 .424-.287.712A.97.97 0 0 1 12 17a.97.97 0 0 1-.713-.288A.97.97 0 0 1 11 16v-4q0-.424.287-.713"
        />
        <path
          clip-rule="evenodd"
          d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10m-2 0a8 8 0 1 1-16 0 8 8 0 0 1 16 0"
          fill-rule="evenodd"
        />
      </svg>
      First item
    </li>
    <li
      class="_visual-list-item_1553df"
    >
      <svg
        aria-hidden="true"
        class="_visual-list-item-icon_1553df"
        fill="currentColor"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.288 7.288A.97.97 0 0 1 12 7q.424 0 .713.287Q13 7.576 13 8t-.287.713A.97.97 0 0 1 12 9a.97.97 0 0 1-.713-.287A.97.97 0 0 1 11 8q0-.424.287-.713m.001 4.001A.97.97 0 0 1 12 11q.424 0 .713.287.287.288.287.713v4q0 .424-.287.712A.97.97 0 0 1 12 17a.97.97 0 0 1-.713-.288A.97.97 0 0 1 11 16v-4q0-.424.287-.713"
        />
        <path
          clip-rule="evenodd"
          d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10m-2 0a8 8 0 1 1-16 0 8 8 0 0 1 16 0"
          fill-rule="evenodd"
        />
      </svg>
      Second item
    </li>
    <li
      class="_visual-list-item_1553df"
    >
      <svg
        aria-hidden="true"
        class="_visual-list-item-icon_1553df"
        fill="currentColor"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.288 7.288A.97.97 0 0 1 12 7q.424 0 .713.287Q13 7.576 13 8t-.287.713A.97.97 0 0 1 12 9a.97.97 0 0 1-.713-.287A.97.97 0 0 1 11 8q0-.424.287-.713m.001 4.001A.97.97 0 0 1 12 11q.424 0 .713.287.287.288.287.713v4q0 .424-.287.712A.97.97 0 0 1 12 17a.97.97 0 0 1-.713-.288A.97.97 0 0 1 11 16v-4q0-.424.287-.713"
        />
        <path
          clip-rule="evenodd"
          d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10m-2 0a8 8 0 1 1-16 0 8 8 0 0 1 16 0"
          fill-rule="evenodd"
        />
      </svg>
      Third item
    </li>
  </ul>
</div>
`;
